skbio/metadata/_intersection.pyx

```diff
@@ -186,7 +186,15 @@ cdef class IntervalNode:
         if self.cleft is not EmptyNode and self.cleft.maxend > start:
             self.cleft._intersect( start, end, results )
         # This interval
-        if ( self.end > start ) and ( self.start < end ):
+        if self.end == self.start:
+            send = self.end
+        else:
+            send = self.end - 1
```

Comment:
I don't get this. what comment you would like me to add here? Originally I didn't think about mimicing the old behaviour...